src/core/map.cpp

```diff
@@ -989,6 +989,12 @@ void Map::drawSelection(QPainter* painter, bool force_min_size, MapWidget* widge
 void Map::addObjectToSelection(Object* object, bool emit_selection_changed)
 {
 	Q_ASSERT(!isObjectSelected(object));
+
+	// we omit hidden and protected objects from any kind of selection
+	const auto object_symbol = object->getSymbol();
```

Comment:
As stated in the other PR, I would prefer `auto*` for pointers.

Reply:
Fixed in code. Thanks for the review!